sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Utilities.cs

```diff
@@ -98,17 +99,17 @@ public static HttpResponseMessage BuildErrorResponse(EventErrorResponse error)
             }
             else
             {
-                // JObject or string type, use string to convert between JObject and JsonDocument.
+                // JObject or string type, use string to convert between NewtonsoftJson and SystemTextJson.
                 originStr = response.ToString();
-                converted = JsonDocument.Parse(originStr);
+                jResponse = response is JObject jObj ? jObj : JObject.Parse(originStr);
             }
 
             try
             {
                 // Check error, errorCode is required for json convert, otherwise, ignored.
-                if (needConvert && converted.RootElement.TryGetProperty("code", out var code))
+                if (needConvert && jResponse.TryGetValue("code", out var code))
                 {
-                    var error = JsonSerializer.Deserialize<EventErrorResponse>(originStr);
+                    var error = SystemJson.JsonSerializer.Deserialize<EventErrorResponse>(originStr);
```

Comment:
Original motivation is because the request/response objects are defined under `Common` project and custom converters are defined under `System.Text.Json` as well. We don't want to add another sets of converters under `Newtonsoft` for easier maintenance unless they're using different logic like `ConnectionStates`.
Review again and just notice responses don't have custom converters then add [DataMember] should be able to convert correctly, so no need to switch Framework here anymore. Let me update.